osquery/events/eventpublisherplugin.cpp

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <chrono>
+#include <ctime>
+#include <exception>
+#include <mutex>
+#include <thread>
+
+#include <boost/algorithm/string.hpp>
+#include <boost/algorithm/string/classification.hpp>
+#include <boost/lexical_cast.hpp>
+
+#include <osquery/config/config.h>
```

Comment:
My last bit of feedback, then this PR looks good: this list of includes seems too broad. Do we really need config, flags, logger, sql, database, etc here? We only include what we use in the CPP file.

Reply:
I've checked the includes and they seem to be correct:

`#include <osquery/config/config.h>`
  `kExecutingQuery`

`#include <osquery/core/flags.h>`
  `FLAG() macro`

`#include <osquery/database/database.h>`
  `scanDatabaseKeys()`

`#include <osquery/events/eventfactory.h>`
  `EventFactory::ForwardEvent()`

`#include <osquery/events/eventpublisher.h>`
  `getPublisher()->removeSubscriptions(getName());`

`#include <osquery/events/eventsubscriberplugin.h>`
  `EventSubscriberPlugin class`

`#include <osquery/logger/logger.h>`
  `LOG() macro`

`#include <osquery/registry/registry_factory.h>`
  `CREATE_REGISTRY() macro`

`#include <osquery/sql/dynamic_table_row.h>`
  `yield(TableRowHolder(new DynamicTableRow(std::move(row))));`

`#include <osquery/utils/conversions/tryto.h>`
  `tryTo<> utilities`

`#include <osquery/utils/system/time.h>`
  `getUnixTime()`

`#include <osquery/utils/info/tool_type.h>`
  `isDaemon()`